spec/datadog_spec.rb

```diff
@@ -792,56 +809,87 @@
       @run_status.stop_clock
 
       @run_status.run_context = @run_context
-
-      # Run the report
-      handler.run_report_unsafe(@run_status)
     end
 
     context 'emits metrics' do
+
       it 'reports metrics' do
-        expect(a_request(:post, METRICS_ENDPOINT).with(
-          :query => { 'api_key' => API_KEY }
-        )).to have_been_made.times(5)
+        dogs = @handler.instance_variable_get(:@dogs)
+
+        # Check that we do have two Dogapi clients (one for each endpoint)
+        expect(dogs.length).to eq(2)
+
+        for dog in dogs do
+          # Define mock functions to avoid failures when connecting to the app.example.com endpoint
+          def dog.emit_point metric, value, options= {}
+            true
+          end
+          def dog.emit_event event, options= {}
+            [200, "{'event': 'My event'}"]
+          end
+          def dog.update_tags host_id, tags, source=nil
+            [201, "Created"]
+          end
+        end
+
+        expect(dogs[0]).to receive(:emit_point).exactly(5).times.and_call_original
```

Comment:
Choosing one assertion per test is better in this case as it would allow to find the problem quicker. Even if boilerplate is repeated. (though you can probably generalize it using `before` and `context`)